File: darshan/week-1/08-19-25 java/5-calling-constructor-using-this/Constructor.java
class Student{

	String name;
	int age;

	Student(){

		this("darshan",23);

		//DEFAULT CONSTRUCTOR
		System.out.println("=== In default constructor ===");
		displayDetails();
		
	} 

	Student(String name, int age){
		//PARAMETERISED CONSTRUCTOR
		System.out.println("=== In parameterised constructor ===");
		this.name = name;
		this.age = age;
	}

	public void displayDetails(){
		System.out.println(name);
		System.out.println(age);

	}

}
class Constructor{

	public static void main(String args[]){
		
		//Invokes default constructor.
		Student s1 = new Student();				
		
	}
}

File: darshan/week-3/09-05-25 dart/3-constructor and types/constructor.dart
void main() {
  Student student = Student();
  print(student.hashCode);
  student.displayStudent();
}

class Student {
  String? name;
  int? rollNo;
  int? age;
  String? subject;

  Student() {
    this.name = "Darshan";
    this.rollNo = 1000;
    this.age = 23;
    this.subject = "Dart programming";
    print(this.hashCode);
  }

  void displayStudent() {
    print("$name ($rollNo) is of $age years, and studies $subject.");
  }
}
